File: Programing_Basics_with_JS/examPreparation/programmingBasicsOnlineExam20And21April2019/02.EasterParty.js
function easterParty(input) {
    let guest = Number(input[0]);
    let priceForOnePerson = Number(input[1]);
    let budget = Number(input[2]);
    
    let cakePrice = budget * 0.10;

    if(guest >= 10 && guest <= 15) {
        priceForOnePerson = priceForOnePerson * 0.85;
    } else if (guest > 15 && guest <= 20) {
        priceForOnePerson = priceForOnePerson * 0.80;
    } else if (guest > 20) {
        priceForOnePerson = priceForOnePerson * 0.75;
    }

    let allEnvelopesPrice = guest * priceForOnePerson;
    let totalCost = allEnvelopesPrice + cakePrice;  
    let moneyDiff = Math.abs(budget - totalCost);

    if(budget >= totalCost) {
        console.log(`It is party time! ${moneyDiff.toFixed(2)} leva left.`);
    } else {
        console.log(`No party! ${moneyDiff.toFixed(2)} leva needed.`);
    }
}

easterParty(["18",
"30",
"450"])

easterParty(["8",
"25",
"340"])

easterParty(["24",
"35",
"550"])

File: Programming_Basics_with_Python/programming_basics_online_exam_20_and_21_april_2019/02._easter_party.py
guest_numbers = int(input())
price_per_person = float(input())
budged = float(input())

cake_price = budged * 0.10
all_person_fee = 0
if 10 <= guest_numbers <= 15:
    price_per_person = price_per_person * 0.85
    all_person_fee = price_per_person * guest_numbers
elif 15 < guest_numbers <= 20:
    price_per_person = price_per_person * 0.80
    all_person_fee = price_per_person * guest_numbers
elif guest_numbers > 20:
    price_per_person = price_per_person * 0.75
    all_person_fee = price_per_person * guest_numbers
else:
    all_person_fee = price_per_person * guest_numbers

total_cost = all_person_fee + cake_price
money_difference = budged - total_cost
if budged >= total_cost:
    print(f'It is party time! {money_difference:.2f} leva left.')
else:
    print(f'No party! {abs(money_difference):.2f} leva needed.')